Adds machine files for Compy

singhbalwinder opened the pull request:
This PR adds first set of machine files for E3SM machine Compy. Machine
files for compilers Intel, gnu and PGI are added. I have used the
following test to compile and run the model:

SMS.ne4_ne4.FC5AV1C-L

For the above test:
-Intel compiler works fine
-GNU 4.8.5: Land model has a compile time error
-PGI: Netcdf is not built using this compiler yet

[BFB]

rljacob commented:
You could just remove the gnu entry and continue with this PR.  Then add 7.3 when its ready.

singhbalwinder commented:
@rljacob : PGI (18.10) and Intel now works fine. MKL library paths have been moved around which I have fixed in the machine files. I have also simplified the machine files a bit. With GNU (8.1.0) compiler, I am getting a run time error (model builds fine):
```
set_time_float_from_date: error return from ESMF_TimeSet for set_time_float_from_date
 ERROR: CHKRC
```
I do not know how to resolve this error.

PGI compiler reported a compile time error due to missing `isnan` use statement in one of the land model's file (`components/clm/src/data_types/VegetationDataType.F90`). I have fixed that and I will issue a separate PR for that fix.

minxu74 commented:
@singhbalwinder Could you try to build the test SMS.T62_oQU120_ais20.MPAS_LISIO_TEST on Compy with PGI? I got a build error with PGI on Summit (#2862 ) and want to see if the same error happened on Compy. Thanks.

singhbalwinder commented:
@minxu74 : I tried this test case but it blew up during setup due to lack of partition files. I think it is most likely due to the default PE layout the model is picking up. I do not know enough about MPAS model to figure out a PE layout which will ask for a "right" PE layout so that the existing files are downloaded automatically from the server.

minxu74 commented:
@singhbalwinder Thanks.  I do not know if you can set the PE layout of the test as the one on Summit. It sets the NTASKS to 64 except for ESP and NTHRDS to 1.  The files needed by MPAS for that PE layout should be in the E3SM input directory or LCRC public web server.

singhbalwinder commented:
@minxu74 : It is using 64 tasks on Compy as well. Right now, I got a build error possibly due to lack on PNETCDF. I have asked the support to install PNETCDF. I will see if I can move forward after they install it.

rljacob commented:
@minxu74 the priority now is just get Compy working.  Diagnosing pgi problems from other machines can wait.

minxu74 commented:
@singhbalwinder @rljacob Sorry, please go ahead and focus on the PR. Thanks

singhbalwinder commented:
@rljacob : I have run few more simulations and everything looks fine with Intel and PGI compilers. MPAS test works with intel but the Lustre drive crashed while I was doing the PGI test. Tim Carlson is currently working on fixing that.

GNU compiler has a run time error which seems related to some timing/calendar issue. I am not sure how to fix that. Should we merge this PR as is or do you think we should remove machine files for GNU and then merge it?

rljacob commented:
Remove the gnu entry, update the branch and then merge.

singhbalwinder commented:
I have removed GNU from the machine files.

rljacob commented:
The keep things simple, lets just focus on the Intel compiler.  Once that is working, more people can get on and do work.

rljacob commented:
Leave PGI in, but don't worry if things aren't building/passing with it.

singhbalwinder commented:
Ok, Intel is working fine with `FC5AV1C-L` with both ne4 and ne30. It also worked fine for the `MPAS_LISIO_TEST` compset with  `T62_oQU120_ais20` grid. I also ran a long 6 year simulation with ne30, F compset and it ran fine. So, I think, this PR is ready to merge once next and master are open to accept new PRs.

rljacob commented:
Since this is machine-specific, you can go ahead and merge to next and master.

singhbalwinder commented:
Thanks @rljacob ! I think it needs your approval before @bibiraju can merge it.
